File: lib/to_mdx/parsers/open_document/dc/title.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module DublinCore
      class Title < Base
        attr_reader :title_text

        def initialize(element = nil, context = {})
          super(element, context)
          @title_text = element&.text&.strip
          context["title"] = @title_text unless @title_text&.empty?
        end

        def to_mdx
          ""
        end

        def to_s
          title_text || ""
        end

        register_for "dc:title"
      end
    end
  end
end

File: src/parsers/open-document/dc/title.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class Title extends BaseElement {
  private titleText: string;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);

    this.titleText = (element.textContent || "").trim();
    if (this.titleText) {
      context.title = this.titleText;
    }
  }

  toMdx(): string {
    return "";
  }

  toString(): string {
    return this.titleText || "";
  }

  static {
    BaseElement.registerFor.call(this, "dc:title");
  }
}
